library/ziti-socket.c

```diff
@@ -187,16 +189,66 @@ static void on_ctx_event(ziti_context ztx, const ziti_event_t *ev) {
                 free(wrap);
             }
         }
+    } else if (ev->type == ZitiServiceEvent) {
+
+
+        for (int i = 0; ev->event.service.removed && ev->event.service.removed[i] != NULL; i++) {
+            ziti_intercept_cfg_v1 *intercept = model_map_remove(&wrap->intercepts, ev->event.service.removed[i]->name);
+            free_ziti_intercept_cfg_v1(intercept);
+            FREE(intercept);
```

Comment:
it's guaranteed that an event will never have removed/changed/added entries, right? otherwise this will double FREE(intercept). Just making sure that's a guarantee...

Reply:
ugh. tired... i see it is scoped in the loop... my bad